### ISSUE TITLE
’--reuse-session' can not work when using '-n NUM'.

### ISSUE THREAD

**Issue by clobob** [seleniumbase/SeleniumBase#563]
hello,
Here is my scenario: 
 1.  create many tests which need re-use 'login' session .
 2.  to run all these tests with option both '--user-dir-data --reuse-session' and '-n NUM'

result:
  all tests run under multiple threads , but ' --reuse-session' does not work.

**Comment by mdmintz** [comment]
Hi @clobob , thank you for reporting this issue. I'll make sure that when using multi-threaded tests, each thread can reuse the browser session. The fix will likely be part of the next big release, which will include the ability to translate tests into multiple languages.

If you're not already familiar with the multi-language support, here's an example test in Chinese: https://github.com/seleniumbase/SeleniumBase/blob/master/examples/translations/chinese_test_1.py
```python
from seleniumbase.translate.chinese import 硒测试用例

class 我的测试类(硒测试用例):

    def test_例子1(self):
        self.开启网址("https://xkcd.in/comic?lg=cn&id=353")
        self.断言标题("Python - XKCD中文站")
        self.断言元素("#content div.comic-body")
        self.断言文本("上漫画")
        self.单击("div.nextLink")
        self.断言文本("老妈的逆袭", "#content h1")
        self.单击链接文本("下一篇")
        self.断言文本("敲桌子", "#content h1")
        self.断言文本("有时候无聊就是最棒的乐趣")
        self.回去()
        self.单击链接文本("兰德尔·门罗")
        self.断言文本("兰德尔·门罗", "#firstHeading")
        self.更新文本("#searchInput", "程式设计")
        self.单击("#searchButton")
        self.断言文本("程序设计", "#firstHeading")
```
When the translation feature is ready, any test can be instantly converted to any language because every method will have a one-to-one mapping to all 10 supported languages. Example: ``self.开启网址(URL)`` works the same as ``self.open(URL)``. All my time is being spent on this new feature right now. It will likely ready in the next 2 or 3 days, which will include the fix for ``--reuse-session``. It should be ready by the weekend.

**Comment by clobob** [comment]
thanks for your reply.
i have another question about '--user-dir-data' and '--reuse-session'. 
here it is:  
can i  use both **options** to affect selenium grid nodes ?

**Comment by mdmintz** [comment]
@clobob  That part should already be possible, at least if you're using your own Selenium Grid. A remote grid may require specific desired capabilities to be set. I have a ReadMe file on setting desired capabilities with SeleniumBase: https://github.com/seleniumbase/SeleniumBase/blob/master/examples/capabilities/ReadMe.md You might not need it though. If you haven't used the SeleniumBase Grid controls yet, here is the ReadMe for that: https://github.com/seleniumbase/SeleniumBase/blob/master/seleniumbase/utilities/selenium_grid/ReadMe.md

**Comment by clobob** [comment]
> @clobob That part should already be possible, at least if you're using your own Selenium Grid. A remote grid may require specific desired capabilities to be set. I have a ReadMe file on setting desired capabilities with SeleniumBase: https://github.com/seleniumbase/SeleniumBase/blob/master/examples/capabilities/ReadMe.md You might not need it though. If you haven't used the SeleniumBase Grid controls yet, here is the ReadMe for that: https://github.com/seleniumbase/SeleniumBase/blob/master/seleniumbase/utilities/selenium_grid/ReadMe.md

My point is that can I use the options '--user-dir-data',  '--reuse-session' and '--server=MyGridServer' in one test-running.

here is my way to execute test:
pytest my_test_suit.py --user-dir-data --reuse-session --server=MyGridServer

I have tried with seleniumbase grid, but user-dir-data does not work. Is there a way to config grid nodes to work for this ?

**Comment by mdmintz** [comment]
@clobob You might be able to with a Docker-based Selenium Grid. I read about it here: https://github.com/SeleniumHQ/docker-selenium/issues/741
It might be easier to not use a Selenium Grid at all if you don't have to.

**Comment by clobob** [comment]
I'll try it. thanks

**Comment by mdmintz** [comment]
@clobob Fixed in SeleniumBase ``1.38.0`` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v1.38.0
